nixos/modules/services/cluster/hadoop/hdfs.nix

```diff
@@ -34,34 +37,54 @@ in
       };
     };
     datanode = {
-      enabled = mkOption {
+      enable = mkEnableOption "Whether to run the HDFS DataNode";
+      inherit restartIfChanged;
+      openFirewall = mkOption {
         type = types.bool;
-        default = false;
+        default = true;
         description = ''
-          Whether to run the HDFS DataNode
+          Open firewall ports for datanode
         '';
       };
+    };
+    journalnode = {
+      enable = mkEnableOption "Whether to run the HDFS JournalNode";
       inherit restartIfChanged;
       openFirewall = mkOption {
         type = types.bool;
         default = true;
         description = ''
-          Open firewall ports for datanode
+          Open firewall ports for journalnode
```

Comment:
Which ones?

Reply:
See line 137 for ports. If you are suggesting that the option description should list opened ports, please see other modules for the current convention of the description of "openFirewall" options.